sunpy/net/jsoc/jsoc.py

```diff
@@ -38,9 +38,14 @@ def __init__(self, table=None):
         table : `astropy.table.Table`
         """
 
-        self.table = table
-        self.query_args = None
-        self.requests = None
+        if isinstance(table, list) and isinstance(table[0], type(self)):
```

Comment:
What happens if it's a list and longer than length 1? Also what happens if it's passed an instance of `type(self)` not in a list?

Reply:
I believe these cases will never arise. The list of length 1 is only created when we handle slicing in Fido.
 https://github.com/sunpy/sunpy/blob/c6b475659c1c366c574e8b755e7a828c31b6182f/sunpy/net/fido_factory.py#L94-97

I'm not sure still of these corner cases appearing in any scenario. Let me know if there is one.

Reply:
OK, I think this is fine as a bugfix, the "correct" fix is probably in #3770 but is much deeper and affects all clients. :+1: